File: src/main/java/org/mifos/fineract/services/CacheApi.java
package org.mifos.fineract.services;

import org.mifos.fineract.models.GetCachesResponse;
import org.mifos.fineract.models.PutCachesRequest;
import org.mifos.fineract.models.PutCachesResponse;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.PUT;
import rx.Observable;

import java.util.List;

public interface CacheApi {
    /**
     * Retrieve Cache Types
     * Returns the list of caches.  Example Requests:  caches
     *
     * @return Observable&lt;List&lt;GetCachesResponse&gt;&gt;
     */
    @Headers({
            "Content-Type:application/json"
    })
    @GET("caches")
    Observable<List<GetCachesResponse>> retrieveAll();


    /**
     * Switch Cache
     * Switches the cache to chosen one.
     *
     * @param body body (required)
     * @return Observable&lt;PutCachesResponse&gt;
     */
    @Headers({
            "Content-Type:application/json"
    })
    @PUT("caches")
    Observable<PutCachesResponse> switchCache(
            @retrofit2.http.Body PutCachesRequest body
    );

}

File: src/main/kotlin/org/openapitools/client/apis/CacheApi.kt
package org.openapitools.client.apis

import org.openapitools.client.models.GetCachesResponse
import org.openapitools.client.models.PutCachesRequest
import org.openapitools.client.models.PutCachesResponse
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.PUT

interface CacheApi {
    /**
     * Retrieve Cache Types
     * Returns the list of caches.  Example Requests:  caches
     * Responses:
     *  - 200: OK
     *
     * @return [kotlin.collections.List<GetCachesResponse>]
     */
    @GET("v1/caches")
    suspend fun retrieveAll4(): List<GetCachesResponse>

    /**
     * Switch Cache
     * Switches the cache to chosen one.
     * Responses:
     *  - 200: OK
     *
     * @param putCachesRequest
     * @return [PutCachesResponse]
     */
    @PUT("v1/caches")
    suspend fun switchCache(@Body putCachesRequest: PutCachesRequest): PutCachesResponse

}
